include/revng/ABI/AArch64/AAPCS64.h

```diff
@@ -70,7 +71,15 @@ struct Trait<model::ABI::AAPCS64> {
 
   static constexpr bool CalleeIsResponsibleForStackCleanup = false;
   static constexpr size_t StackAlignment = 16;
-  static constexpr size_t MinimumStackArgumentSize = 8;
+  static constexpr std::array ScalarStackAlignment{
+    abi::Alignment{ .Size = 1, .PadTo = 1, .AlignAt = 1 },
```

Comment:
So `.Size` is the key of the data structure, but `PadTo` and `AlignAt`? Are they documented somewhere?
Intuitively `PadTo` is a consequence of `AlignAt`, but this doesn't seem to be the case.

Reply:
I really didn't want to split the two. But there are cases where it's non-obvious (for example under `s390x`, it's acceptable to pad 16-byte types to 16 bytes but they are still aligned on 8-byte boundaries.

Reply:
What you mean by "acceptable"? The ABI should prescribe one and one way only to do things...

Can you link the source?
Share some pain with me.

Reply:
Bad wording on my part. I meant that there are types with size bigger than the alignment - hence it's necessary to have a way to preserve both.

(I'll send the source on mattermost, since I built the pdf from the whatever sources (probably latex) they used in their github repo).

Reply:
As per discussion on mattermost, and recent redesign, this thread is no longer relevant.